experiment/manage_release_versions.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python
+
+# Copyright 2019 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+""" This script rotates the version strings in various below listed config files.
+Run this script when you want to move the jobs to the new support cadence.
+
+Usage:
+    1. update the Version Class below with the appropriate versions.
+    2. bazel run //experiment:manage_release_versions
+    3. for some config files you may need to make more manual changes. These
+    are described as comments before config files are defined below.
+
+"""
+
+import re
+import collections
+
+# Probably should just read out of a config file in the future.
+
+
+class Version(object):
+    CURRENT = "1.13"
+    STABLE1 = "1.12"
+    STABLE2 = "1.11"
+    STABLE3 = "1.10"
+
+# ordered-dict that keeps the mapping of the rotations.
+# we need a ordered dict to make sure we rotate in the
+# order of latest to oldest.
+RELEASE_VERSION_ROTATION_MAP = collections.OrderedDict([
+    (Version.STABLE1, Version.CURRENT),
+    (Version.STABLE2, Version.STABLE1),
+    (Version.STABLE3, Version.STABLE2),
+])
+# List of job config locations that need version rotation.
+# We might have to do more than just rotation in these job configs.
+# Assuming that the script is executed from the test-infra root
+# directory when run directly.
+#
+
+JOB_LIST = [
+    "README.md",
+    "config/jobs/kubernetes/sig-aws/kops/kops-presubmits.yaml",
+    "config/jobs/kubernetes-security/generated-security-jobs.yaml",
```

Comment:
let's also leave out generated jobs, these are created by `hack/update-config.sh` .

Reply:
So when is the hack/update-config.sh run? Who runs it?

Reply:
Right now we have a presubmit job that makes sure that the PR author has run `hack/update-config.sh` manually. The idea there is that the generated security jobs are derived from the existing jobs. We could either deprecate `hack/update-config.sh` and port over all the logic here or call it from inside this script. Let's leave that as a follow up and comment out this config for now.